eZ/Bundle/EzPublishCoreBundle/Imagine/Variation/AliasGeneratorDecorator.php

```diff
@@ -0,0 +1,148 @@
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Imagine\Variation;
+
+use eZ\Bundle\EzPublishCoreBundle\Imagine\IORepositoryResolver;
+use eZ\Bundle\EzPublishCoreBundle\Imagine\VariationPathGenerator;
+use eZ\Publish\API\Repository\Values\Content\Field;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+use eZ\Publish\Core\IO\IOServiceInterface;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\MVC\Exception\SourceImageNotFoundException;
+use eZ\Publish\SPI\Variation\Values\ImageVariation;
+use eZ\Publish\SPI\Variation\VariationHandler;
+use Imagine\Image\ImagineInterface;
+
+/**
+ * Alias Generator Decorator which ensures (using Imagine if needed) that ImageVariation has proper
+ * dimensions.
+ */
+class AliasGeneratorDecorator implements VariationHandler
+{
+    /**
+     * @var \eZ\Publish\SPI\Variation\VariationHandler
+     */
+    private $aliasGenerator;
+
+    /**
+     * @var \eZ\Bundle\EzPublishCoreBundle\Imagine\VariationPathGenerator
+     */
+    private $variationPathGenerator;
+
+    /**
+     * @var \eZ\Publish\Core\IO\IOServiceInterface
+     */
+    private $ioService;
+
+    /**
+     * @var \eZ\Publish\Core\MVC\ConfigResolverInterface
+     */
+    private $configResolver;
+
+    /**
+     * @var \Imagine\Image\ImagineInterface
+     */
+    private $imagine;
+
+    /**
+     * Web Root directory.
+     *
+     * @var string
+     */
+    private $webRootDir;
+
+    public function __construct(
+        VariationHandler $aliasGenerator,
+        VariationPathGenerator $variationPathGenerator,
+        IOServiceInterface $ioService,
+        ConfigResolverInterface $configResolver,
+        ImagineInterface $imagine,
+        $webRootDir
+    ) {
+        $this->aliasGenerator = $aliasGenerator;
+        $this->variationPathGenerator = $variationPathGenerator;
+        $this->ioService = $ioService;
+        $this->configResolver = $configResolver;
+        $this->imagine = $imagine;
+        $this->webRootDir = $webRootDir;
+    }
+
+    /**
+     * Returns a Variation object, ensuring proper image dimensions.
+     *
+     * {@inheritdoc}
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function getVariation(
+        Field $field,
+        VersionInfo $versionInfo,
+        $variationName,
+        array $parameters = []
+    ) {
+        /** @var \eZ\Publish\SPI\Variation\Values\ImageVariation $variation */
+        $variation = $this->aliasGenerator->getVariation(
+            $field,
+            $versionInfo,
+            $variationName,
+            $parameters
+        );
+
+        if (null === $variation->width || null === $variation->height) {
+            $variationBinaryFile = $this->getVariationBinaryFile($field->value->id, $variationName);
+            $absoluteVariationPath = realpath($this->webRootDir . $variationBinaryFile->uri);
```

Comment:
I'm sorry, but this is not working with DFS.

Please, consider a change to something like this:
```
$variationBinaryFile = $this->getVariationBinaryFile($field->value->id, $variationName);
$image = $this->imagine->load($this->ioService->getFileContents($variationBinaryFile));
$dimensions = $image->getSize();

return new ImageVariation(
...
```
Thanks to that IOService will be responsible for finding and loading a proper file using right adapter.
Then, you can remove `$webRootDir` variable.